vividus-plugin-web-app/src/main/java/org/vividus/bdd/steps/ui/web/SetContextSteps.java

```diff
@@ -291,58 +298,84 @@ public void switchingToWindow()
     }
 
     /**
-     * Switch the focus of future browser commands to the new <b>window object</b> with the specified part of
-     * <b>windowName</b>.
-     * <p>
-     * Each browser <b>window</b> or <b>tab</b> is considered to be a separate <b>window object</b>. This object holds
-     * corresponding <b>Document</b> object, which itself is a html page.
-     * </p>
-     * <b>WindowName</b> references to the page title, which is set by {@code <title>} tag.
-     * <p>
-     * Actions performed at this step:
-     * <ul>
-     * <li>Searches among currently opened windows (and tabs) for a window which has the specified part of the
-     * <b>windowName</b>.</li>
-     * <li>If such window is found switches focus to it. If window is not found current focus stays unchanged;</li>
-     * </ul>
-     * @param windowPartName Any part of the value of the {@code <title>} tag of a desired window
-     * @see <a href="https://html.spec.whatwg.org/#browsing-context"><i>Browsing context (Window &amp; Document)</i></a>
-     * @see <a href="https://www.w3schools.com/tags/default.asp"><i>HTML Element Reference</i></a>
-     */
-    @When("I switch to a window with the name containing '$windowPartName'")
-    public void switchingToWindowPartName(String windowPartName)
+    * Wait for a window and switches the focus of future browser commands to the new
+    * <b>window object</b> with the specified <b>title</b>.
+    * <p>
+    * Each browser <b>window</b> or <b>tab</b> is considered to be a separate <b>window object</b>. This object holds
+    * corresponding <b>Document</b> object, which itself is a HTML page.
+    * </p>
+    * <b>Title</b> references to the page title, which is set by {@code <title>} tag.
+    * <p>
+    * Actions performed at this step:
+    * <ul>
+    * <li>Searches among currently opened windows (and tabs) for a window with the specified <b>windowName</b>.</li>
+    * <li>If such window is found switches focus to it. If window is not found current focus stays unchanged;</li>
+    * </ul>
+    * @param comparisonRule is equal to, contains, does not contain
+    * @param title Value of the {@code <title>} tag of a desired window
+    * @param duration in format <a href="https://en.wikipedia.org/wiki/ISO_8601">Duration Format</a>
+    * @see <a href="https://html.spec.whatwg.org/#browsing-context"><i>Browsing context (Window &amp; Document)</i></a>
+    * @see <a href="https://www.w3schools.com/tags/default.asp"><i>HTML Element Reference</i></a>
+    */
+    @When("I wait `$duration` until window with title that $comparisonRule `$title` appears and switch to it")
+    public void waitForWindowAndSwitch(Duration duration, StringComparisonRule comparisonRule, String title)
     {
-        switchToWindow(containsString(windowPartName));
+        Matcher<String> expected = comparisonRule.createMatcher(title);
+        WaitResult<Boolean> result = waitActions.wait(webDriverProvider.get(), duration, new ExpectedCondition<>()
```

Comment:
does it makes sense to move out expected condition to separate class as we usually do it?